src/client/home/MintSection/MintingModalButton/MintingModal/MintButton/ConnectButton.tsx

```diff
@@ -2,20 +2,34 @@ import Button from '@/components/buttons/Button/Button';
 import { useAuth } from '@/client/home/useAuth';
 import React from 'react';
 import { useTranslation } from 'react-i18next';
+import DisplayIf from '@/components/conditionals/DisplayIf';
+import Divider from '@/components/dividers/Divider/Divider';
+import { toParsedAddress } from '@/client/home/utils';
+import DisconnectButton from '../WalletDetails/DisconnectButton';
 
 interface ConnectButtonProps {
   className?: string;
 }
 
 const ConnectButton: React.FC<ConnectButtonProps> = ({ className }) => {
-  const { login } = useAuth();
+  const { login, address } = useAuth();
 
   const { t } = useTranslation();
 
   return (
-    <Button onClick={login} className={`w-full border border-gray-500 p-2 bg-gray-900 text-white !rounded-md ${className}`}>
-      {t('Connect Wallet')}
-    </Button>
+    <DisplayIf
+      falsy={
+        <Button onClick={login} className={` border border-gray-500 p-2 bg-gray-900 text-white !rounded-md ${className}`}>
+          {t('Connect Wallet')}
+        </Button>
```

Comment:
I would create a Button inside the component folder and reuse it everywhere, that button can take a title as argument, etc